packages/web/src/components/createCell.tsx

```diff
@@ -111,103 +212,115 @@ const isDataEmptyArray = (data: DataObject) => {
+   */
   if (global.__REDWOOD__PRERENDERING) {
-    // If its prerendering, render the Cell's Loading component
-    // and exit early. The apolloclient loading props aren't available here, so 'any'
+    /**
+     * Apollo Client's props aren't available here, so 'any'.
+     */
     return (props) => <Loading {...(props as any)} />
   }
 
-  const NamedCell = (props: React.PropsWithChildren<CellProps>) => {
-    // destructuring to not pass children to beforeQuery
-    const { children: _children, ...variables } = props
-
-    return (
-      <Query
-        query={
-          typeof QUERY === 'function' ? QUERY(beforeQuery(variables)) : QUERY
-        }
-        {...beforeQuery(variables)}
-      >
-        {({ error, loading, data, ...queryRest }) => {
-          if (error) {
-            if (Failure) {
-              return (
-                <Failure
-                  error={error}
-                  /**
-                   * error code
-                   * @optional
-                   * @type {string}
-                   * @see https://www.apollographql.com/docs/apollo-server/data/errors/#error-codes
-                   * The error code came from `error.graphQLErrors[0].extensions.code`
-                   */
-                  errorCode={
-                    error.graphQLErrors?.[0]?.extensions?.['code'] as string
-                  }
-                  {...{ updating: loading, ...queryRest, ...props }}
-                />
-              )
-            } else {
-              throw error
-            }
-          } else if (data) {
-            if (
-              typeof Empty !== 'undefined' &&
-              isEmpty(data, { isDataEmpty })
-            ) {
-              return (
-                <Empty
-                  {...{ ...data, updating: loading, ...queryRest, ...props }}
-                />
-              )
-            } else {
-              return (
-                <Success
-                  {...afterQuery(data)}
-                  {...{ updating: loading, ...queryRest, ...props }}
-                />
-              )
-            }
-          } else if (loading) {
-            return <Loading {...queryRest} {...props} />
-          } else {
-            console.warn(
-              `If you're using Apollo Client, check for its debug logs here in the console, which may help explain the error.`
-            )
-            throw new Error(
-              'Cannot render Cell: reached an unexpected state where the query succeeded but `data` is `null`. If this happened in Storybook, your query could be missing fields; otherwise this is most likely a GraphQL caching bug. Note that adding an `id` field to all the fields on your query may fix the issue.'
-            )
-          }
-        }}
-      </Query>
+  /**
+   * A Cell declaratively renders components based on the lifecycle of a GraphQL query.
+   *
+   * If the return's confusing, conider that a Cell's usually used in a Page like so:
```

Comment:
```suggestion
   * If the return's confusing, consider that a Cell's usually used in a Page like so:
```

BTW, I couln't understand this specific line. 🤔 Just me, maybe? 😅

Reply:
Nah I think you're right, I was a little too into it. The new react docs have a line "This is why you should not nest component function definitions." I thought for a second that we were doing that but this is different. Happy to get rid of this one!